src/Illuminate/Validation/Validator.php

```diff
@@ -747,7 +747,7 @@ protected function validateDifferent($attribute, $value, $parameters)
 
         $other = Arr::get($this->data, $parameters[0]);
 
-        return isset($other) && $value != $other;
+        return isset($other) && $value !== $other;
```

Comment:
Are you strongly against or should I resend this to 5.2?

To my mind the fact that you can confirm your "1000" password with "1e3" is pretty meh.